frontend/packages/console-app/src/components/nodes/NodeLogs.tsx

```diff
@@ -132,13 +144,19 @@ const NodeLogs: React.FC<NodeLogsProps> = ({ obj: node }) => {
     setPath(newAPI);
     setLogFilenames([]);
     setLogFilename('');
+    setUnit('');
     setQueryArgument(pathQueryArgument, newAPI);
+    removeQueryArgument(unitQueryArgument);
     removeQueryArgument(logQueryArgument);
     setLoadingFilenames(true);
     setLoadingLog(true);
     trimmedContent = '';
   };
   const onTogglePath = () => setPathOpen(!isPathOpen);
+  const onChangeUnit = (value: string) => {
```

Comment:
If the typing occurs too rapidly, query doesn't return the correct results.  This needs a debounce?  @spadgett, @TheRealJon

Reply:
We might want to handle this like label filters on the search page where the user has to click a button or press enter to apply the filter, and then we can show the active units using filter chips. This use case doesn't seem well-suited to an as-you-type filter since getting the logs can be expensive and there's no substring match.